### PR TITLE
[Merged by Bors] - feat: remove the diamond for Complex.measureSpace

### PR DESCRIPTION
**Pull request by xroblot**
We remove the instance 
```lean
instance measureSpace : MeasureSpace ℂ :=  ⟨Measure.map basisOneI.equivFun.symm volume⟩
```
in `MeasureTheory.Measure.Lebesgue.Complex` since `ℂ` has already a `measureSpace` instance coming from its `InnerProductSpace` structure over `ℝ`, and fix the proof of `volume_preserving_equiv_pi`.

Co-authored-by: Eric Wieser <wieser.eric@gmail.com>

---

- [x] depends on: #7034 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean:50]: Does this work as
```suggestion
instance : Measure.IsAddHaarMeasure (@volume ℂ _) :=
```

Comment (xroblot) [reply Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean]: Yes, it does. Thanks.

**Review by eric-wieser — APPROVED** [review]
LGTM

My only question is whether this is considered a "heavy" import, and if it should go in a new `Haar.Complex` file instead. I'll let another maintainer decide

maintainer merge.

**Comment by sgouezel** [comment]
There is a problem with this file, that we have a diamond: complex numbers already have a measure space instance before the start of the file (you can try `example : MeasureSpace ℂ := by infer_instance` before the instance declaration) -- and this instance is already known to be a Haar measure (try `example : Measure.IsAddHaarMeasure (@volume ℂ _) := by infer_instance`). So instead of adding this new instance, we should fix the file and remove the diamond (by just removing the instance)

**Comment by eric-wieser** [comment]
> There is a problem with this file, that we have a diamond

Indeed, this is discussed [here](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.60measure_space.60.20instance.20on.20euclidean.20space/near/350846952)

**Comment by xroblot** [comment]
I see, so the plan is to remove the instance at the start of the file (then the Haar measure instance is synthesised automatically) and just come up with  a new proof of `volume_preserving_equiv_pi` (and probably a new definition for `measurableEquivPi` first).

File: Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean:213]: I have a more-bundled version of this result [here](https://github.com/leanprover-community/mathlib/pull/19013/files#diff-10dd661e90f9d04ff1ba015bde0648b9e60bec561ebeca5f2576e9e8b8a04e4eR74-R91), which it would be nice to have as well; does the bundling help at all?

File: Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean:259]: Nice, I think this lemma was the one I probably needed in https://github.com/leanprover-community/mathlib/pull/19013. If you fancy a hopefully-easy follow-up PR, it would be great to have that result (I imagine most of my work there is worthless)

File: Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean
Comment (xroblot) [reply Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean:213]: Yes, your version seems better.

File: Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean
Comment (eric-wieser) [reply Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean:259]: > Or I guess I can cherrypick some results from your PR to get this one merge rapidly and then move to #19013?

That sounds good to me; cherry-pick the results into this PR that are both useful for `Complex` and that you think are an improvement over your lemmas, and leave the rest to a later PR.

Comment (xroblot) [reply Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean]: #7037

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**
